lib/msf/core/db_manager/note.rb

```diff
@@ -21,10 +21,24 @@ def find_or_create_note(opts)
   #
   # This methods returns a list of all notes in the database
   #
-  def notes(wspace=workspace)
-  ::ActiveRecord::Base.connection_pool.with_connection {
-    wspace.notes
-  }
+  def notes(opts)
+    wspace = opts.delete(:workspace) || opts.delete(:wspace) || workspace
+    if wspace.kind_of? String
+      wspace = find_workspace(wspace)
```

Comment:
```ruby
wspace = find_workspace(wspace) if wspace.kind_of? String
``` 

as per the [style guide](https://github.com/bbatsov/ruby-style-guide#if-as-a-modifier)

Reply:
For what it's worth, this will be cleaned up when #9859 lands as it adds a helper method to do this same block of logic.